temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java

```diff
@@ -278,9 +283,20 @@ public String getTaskQueue() {
     return null;
   }
 
+  /** Returns name of the task queue that test worker is polling. */
+  public String getTaskQueue() {
+    return taskQueue;
+  }
+
   /** Returns client to the Temporal service used to start and query workflows. */
   public WorkflowClient getWorkflowClient() {
-    return testEnvironment.getWorkflowClient();
+    if (useExtnlService) {
+      return WorkflowClient.newInstance(
+          testEnvironment.getWorkflowClient().getWorkflowServiceStubs(),
+          WorkflowClientOptions.newBuilder().setNamespace(testEnvironment.getNamespace()).build());
+    } else {
+      return testEnvironment.getWorkflowClient();
+    }
```

Comment:
Unless I miss something, this change removes the option to have `TimeLockingInterceptor` with external service via the `TestWorkflowRule`?

It might still be useful in some scenarios, e.g. TestContainers can be used to spin up a "real" Temporal instance that can still benefit from the time-rewinding logic of the `TestWorkflowEnvironment`.

Reply:
The time skipping logic is not really supported when a real service is used. It requires changes to the service itself. We have plans to add such logic to the service. It would allow getting rid of the need to implement per SDK testing servcie.